src/components/Workspace/DeleteAction/index.tsx

```diff
@@ -88,7 +154,9 @@ class WorkspaceDeleteAction extends React.PureComponent<Props, State> {
 
 }
 
-const mapStateToProps = () => ({});
+const mapStateToProps = (state: AppState) => ({
+  allWorkspaces: selectAllWorkspaces(state),
```

Comment:
Why don't you pass a workspace status as component property instead of selecting all workspaces?

Reply:
Because it is a different behavior depends on status.  I am going to reuse this action now. So, I decided to check the status of the widget.

Reply:
That's what I meant. You don't need to have this component connected at all, just pass the workspace status from parent component.